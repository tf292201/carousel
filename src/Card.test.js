

import React from 'react';
import { render } from '@testing-library/react';
import Card from './Card';

//smoke test for Card component
test('renders Card component without crashing', () => {
    const caption = 'Test Caption';
    const src = 'test.jpg';
    const currNum = 1;
    const totalNum = 2;
  
    render(<Card caption={caption} src={src} currNum={currNum} totalNum={totalNum} />);
  });


//snapshot test for Card component

test('renders Card component correctly', () => {
  const caption = 'Test Caption';
  const src = 'test.jpg';
  const currNum = 1;
  const totalNum = 2;

  const { container } = render(<Card caption={caption} src={src} currNum={currNum} totalNum={totalNum} />);
  expect(container.firstChild).toMatchSnapshot();
});
