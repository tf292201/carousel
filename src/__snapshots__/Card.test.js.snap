// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card component correctly 1`] = `
<div
  class="Card"
>
  <h4
    class="Card-title"
  >
    Test Caption
  </h4>
  <img
    alt="Test Caption"
    class="Card-image"
    src="test.jpg"
  />
  <small
    class="Card-small"
  >
    Image 
    1
     of 
    2
    .
  </small>
</div>
`;
